source/common/http/conn_manager_utility.cc

```diff
@@ -75,13 +75,27 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
         connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
   }
 
-  // At this point we can determine whether this is an internal or external request. This is done
-  // via XFF, which was set above or we trust.
-  bool internal_request = Network::Utility::isInternalAddress(*final_remote_address);
+  // At this point we can determine whether this is an internal or external request. The
+  // determination of internal status uses the following:
+  // 1) After remote address/XFF appending, the XFF header must contain a *single* address.
+  // 2) The single address must be an internal address.
+  // 3) If configured to not use remote address, but no XFF header is available, even if the real
+  //    remote is internal, the request is considered external.
```

Comment:
Any docs which should be updated?

Reply:
I will check the docs. I think this is covered but if not I will clarify.

Reply:
I think these docs cover it: https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/headers#x-forwarded-for

LMK if not.